cloudinit/config/cc_ssh_import_id.py

```diff
@@ -56,13 +60,16 @@ def handle(_name, cfg, cloud, log, args):
     for (user, user_cfg) in users.items():
         import_ids = []
         if user_cfg['default']:
-            import_ids = util.get_cfg_option_list(cfg, "ssh_import_id", [])
+            import_ids = cfg.get("ssh_import_id", [])
         else:
-            try:
-                import_ids = user_cfg['ssh_import_id']
-            except Exception:
-                log.debug("User %s is not configured for ssh_import_id", user)
-                continue
+            import_ids = user_cfg.get('ssh_import_id', [])
+
+        if not import_ids:
+            continue
+
+        # handle space separated list
+        if isinstance(import_ids, str):
+            import_ids = import_ids.split(" ")
```

Comment:
Specifying `" "` here means that multiple spaces will result in empty string list entries:

```py
>>> "foo  bar".split(" ")
['foo', '', 'bar']
>>> "foo  bar".split()
['foo', 'bar']
```

and that non-ASCII spaces won't be correctly split on:

```py
>>> "foo\u2009bar".split(" ")
['foo\u2009bar']
>>> "foo\u2009bar".split()
['foo', 'bar']
```

Reply:
> Specifying `" "` here means that multiple spaces will result in empty string list entries:
> 
> ```python
> >>> "foo  bar".split(" ")
> ['foo', '', 'bar']
> >>> "foo  bar".split()
> ['foo', 'bar']
> ```

Empty items won't matter because later that list is passed to util.uniq_merge which drops empties.
But agreed on non-ASCII whitespace.

And yes, this does complicate the json schema definition when written for this module, but it doesn't make it hard to validate, just a bit too "flexible" a schena instead of strict types. As, rharper metnioned, there are platform reasons, so I'm also ok with going this route.